dataactbroker/handlers/fileHandler.py

```diff
@@ -1506,13 +1506,29 @@ def add_list_submission_filters(query, filters):
         Raises:
             ResponseException - invalid type is provided for one of the filters
     """
+    # Checking for submission ID filter
     if 'submission_ids' in filters:
         sub_list = filters['submission_ids']
         if sub_list and isinstance(sub_list, list):
             sub_list = [int(sub_id) for sub_id in sub_list]
             query = query.filter(Submission.submission_id.in_(sub_list))
         elif sub_list:
             raise ResponseException("submission_ids filter must be null or an array", StatusCode.CLIENT_ERROR)
+    # Date range filter
+    if 'start_date' in filters or 'end_date' in filters:
+        # Must provide both start and end date
+        if 'start_date' not in filters or 'end_date' not in filters:
+            raise ResponseException("If either start_date or end_date is provided in filters, both must be provided",
+                                    StatusCode.CLIENT_ERROR)
+        start_date = filters['start_date']
+        end_date = filters['end_date']
+        # Start and end dates must be in the format MM/DD/YYYY and be
+        if not (StringCleaner.is_date(start_date) and StringCleaner.is_date(end_date)):
+            raise ResponseException("Start or end date cannot be parsed into a date of format MM/DD/YYYY",
+                                    StatusCode.CLIENT_ERROR)
+        # Need to add a day to the end date to make it actually inclusive
```

Comment:
Why can't we do a `<=` with the `end_date` provided?

Reply:
Because technically it's `date 0:00:00` so it checks the same day/time twice. If it was the end of the day that'd be fine, but essentially by adding a day we're checking `< next day 0:00:00`

Reply:
Fair enough, adding a day is simpler than adding `23:59`.